godot-macros/src/class/data_models/func.rs

```diff
@@ -238,7 +243,16 @@ fn make_forwarding_closure(
             let method_call = if matches!(before_kind, BeforeKind::OnlyBefore) {
                 TokenStream::new()
             } else {
-                quote! { instance.#method_name( #(#params),* ) }
+                match api_trait {
+                    Some(api_trait) => {
+                        let instance_ref = match signature_info.receiver_type {
+                            ReceiverType::Mut => quote! { &mut instance },
+                            _ => quote! { &instance },
```

Comment:
do we want to just silently fail here if an incorrect receiver is passed somehow? it should generate a compile error either way, but maybe we'd wanna explicitly pass a `compile_error!` instead?

Reply:
This is inside of a match arm against `receiver_type` that matches `ReceiverType::Mut` and `ReceiverType::Ref`, so it is impossible in this case for it to be anything else.

Reply:
i think it would be more clear to write that using an `unreachable!` then

Reply:
Applied.